apps/04TowersOfHanoi.js

```diff
@@ -18,25 +18,40 @@ function printStacks() {
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
-
-function movePiece() {
-  // Your code here
-
+// This code below enables the user to move the peices in the array
+function movePiece(startStack, endStack) {
+  stacks[endStack].push(stacks[startStack].pop());
 }
-
-function isLegal() {
+// This code below checks to see if the users move was legal
+function isLegal(startStack, endStack) {
   // Your code here
+  var startBlock = stacks[startStack][stacks[startStack].length - 1];
+  var endBlock = stacks[endStack][stacks[endStack].length - 1];
+  if (startBlock < endBlock || endBlock === undefined) {
+    movePiece(startStack, endStack);
+    return true;
+  } else if (startBlock > endBlock) {
+    console.log("Try Again");
+    return false;
```

Comment:
Notice that these return values aren't used anywhere (except directly by the tests)